src/cdk/tree/tree.ts

```diff
@@ -251,6 +252,15 @@ export class CdkTree<T, K = T> implements AfterContentChecked, CollectionViewer,
       },
     );
 
+    changes.forEachIdentityChange(record => {
+      if (record.currentIndex !== null) {
+        const viewRef = <EmbeddedViewRef<CdkTreeNodeOutletContext<T>>>(
+          viewContainer.get(record.currentIndex)
+        );
+        viewRef.context.$implicit = record.item;
+      }
+    });
+
```

Comment:
This makes sure to actually update the embedded view when a there is an "identity change" in the differ, similar to `NgFor` (I even looked in that directive for a clue).

This fix is even beneficial for flat (non-nested) trees, which has the same issue.